### PR TITLE
[ksm] make time based metrics comparable.

### PR DESCRIPTION
**Pull request by bjhaid**
### What does this PR do?

Prior to this patch the ksm collector would call
`time.Now().Unix.()-metric.Val` at every point a `_time` metric is
computed, this leads to the values collected in the same loop not being
comparable since their values skews based on the value of `time.Now()`
at the time metric was collected.

For an example of the impact of the above, to compute how long it takes a
pod to start after being created we create a function:

```
avg:kubernetes_state.pod.age{pod_name:filebeat-lf6hz} by {pod_name}
- avg:kubernetes_state.pod.uptime{pod_name:filebeat-lf6hz} by {pod_name}
```

When this is plotted on a dashboard we end up with values fluctuating
between negative and positive.

To ensure the values are comparable the `currentTime` is snapshotted at
the start of the Run loop and the value is reused in all places that had
previously called `time.Now()`.

### Motivation

We need to determine how long our pods take from creation to starting, to
measure control plane overhead. The screenshot below depicts the problem
this patch is attempting to solve:

image

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes/notes/ksm-comparable-metrics-b31f1f1fcbae1753.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  - |
+issues:
+  - |
+deprecations:
+  - |
+security:
+  - |
+fixes:
+  - |
+    [ksm] make time based metrics comparable.
```
Comment (kayayarai) [inline releasenotes/notes/ksm-comparable-metrics-b31f1f1fcbae1753.yaml:23]:
```suggestion
    [ksm] KSM Time-based metrics are now comparable.
```

**Review by kayayarai — CHANGES_REQUESTED** [review]
Please confirm, but I think you need to delete the sections you aren't using ("upgrade" "feature" etc) from the release note template.

File: releasenotes/notes/ksm-comparable-metrics-b31f1f1fcbae1753.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+other:
+  - |
```
Comment (ahmed-mez) [inline releasenotes/notes/ksm-comparable-metrics-b31f1f1fcbae1753.yaml:25]:
```suggestion
---
fixes:
  - |
    Time-based metrics (e.g `kubernetes_state.pod.age`, `kubernetes_state.pod.uptime`) are now comparable in the Kubernetes state core check.
```

**Review by ahmed-mez — COMMENTED** [review]
Thank you for the contribution @bjhaid ! The code change looks good.

The Datadog Cluster Agent can run the KSM Core check, could you also copy the release note file in the `releasenotes-dca/notes` folder as well? 

Thanks!

File: releasenotes/notes/ksm-comparable-metrics-b31f1f1fcbae1753.yaml (hunk 1 above)
Comment (bjhaid) [reply releasenotes/notes/ksm-comparable-metrics-b31f1f1fcbae1753.yaml]: thanks! addressed, along with @ahmed-mez 's feedback below

File: releasenotes/notes/ksm-comparable-metrics-b31f1f1fcbae1753.yaml (hunk 2 above)
Comment (bjhaid) [reply releasenotes/notes/ksm-comparable-metrics-b31f1f1fcbae1753.yaml]: thanks! addressed

**Review by ahmed-mez — APPROVED** [review]
> do you have an estimate on when the release schedule for v1.20 is? Thanks!

We're actively working on releasing v1.19 right now - v1.20 is expected to be released in ~2 months.

Thanks for the contribution!

**Comment by bjhaid** [comment]
@kayayarai PTAL, thanks!

File: releasenotes-dca/notes/ksm-comparable-metrics-b31f1f1fcbae1753.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    Time-based metrics (e.g `kubernetes_state.pod.age`, `kubernetes_state.pod.uptime`) are now comparable in the Kubernetes state core check.
```
Comment (kayayarai) [inline releasenotes-dca/notes/ksm-comparable-metrics-b31f1f1fcbae1753.yaml:11]:
```suggestion
    Time-based metrics (for example, `kubernetes_state.pod.age`, `kubernetes_state.pod.uptime`) are now comparable in the Kubernetes state core check.
```

File: releasenotes/notes/ksm-comparable-metrics-b31f1f1fcbae1753.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    Time-based metrics (e.g `kubernetes_state.pod.age`, `kubernetes_state.pod.uptime`) are now comparable in the Kubernetes state core check.
```
Comment (kayayarai) [inline releasenotes/notes/ksm-comparable-metrics-b31f1f1fcbae1753.yaml:11]:
```suggestion
    Time-based metrics (for example, `kubernetes_state.pod.age`, `kubernetes_state.pod.uptime`) are now comparable in the Kubernetes state core check.
```

**Review by kayayarai — APPROVED** [review]
An edit suggestion to change the latinate abbreviation to English. Otherwise 👍  for docs.

**Comment by bjhaid** [comment]
@ahmed-mez @kayayarai thanks a lot for your reviews, very much appreciated!

**Comment by bjhaid** [comment]
@ahmed-mez do you have an ETA for v1.20?

**Comment by ahmed-mez** [comment]
Hey @bjhaid Cluster Agent 1.20 was released last week

**Pull request merged**